R/utils.R

```diff
@@ -372,16 +372,74 @@ markdown_to_text <- function(text) {
     unname()
 }
 
-# Get prepending and appending text based on a simple pattern
-get_pre_post_txt <- function(pattern) {
+#' Handle formatting of a pattern in a \code{fmt_*()} function
+#'
+#' Within the context of a \code{fmt_*()} function, we always have the
+#' single-length character vector of \code{pattern} available to describe a
+#' final decoration of the formatted values. We use \pkg{glue}'s semantics here
+#' (and literally, it's \code{glue::glue()}) and reserve \code{x} to be the
+#' formatted values. We can use \code{x} multiple times in the pattern or even
+#' variables in global scope or R statements within curly braces.
+#' @param pattern An expression string that is the primary input to
+#'   \code{glue::glue()}). The variable \code{x} is reserved and represents the
+#'   formatted values.
+#' @param values The values (as a character vector) that are formatting within
+#'   the \code{fmt_*()} function.
+#' @importFrom glue glue
+#' @noRd
+apply_pattern_fmt_x <- function(pattern,
+                                values) {
 
-  prefix <- strsplit(pattern, "\\{x\\}")[[1]][1]
-  suffix <- strsplit(pattern, "\\{x\\}")[[1]][2]
+  glue::glue(pattern, x = values) %>% as.character()
```

Comment:
Do you really want glue here? It will mean that patterns can be used to execute arbitrary R code. It'd be more conservative to just look for, and replace, `{x}`.

Reply:
This is something I wasn't entirely sure about and so I wanted your opinion on the safety aspect. Using glue seems very powerful because of allowing execution within `{ }` but that's also the disadvantage.

I'm thinking now that the original, non-glue plan might be better.